monai/networks/layers/simplelayers.py

```diff
@@ -41,38 +42,39 @@ def __init__(self, spatial_dims, sigma, truncated=4.0, device=None):
         Args:
             spatial_dims (int): number of spatial dimensions of the input image.
                 must have shape (Batch, channels, H[, W, ...]).
-            sigma (float): std.
+            sigma (float or sequence of floats): std.
             truncated (float): spreads how many stds.
             device (torch.device): device on which the tensor will be allocated.
         """
-        self.kernel = torch.nn.Parameter(torch.tensor(gaussian_1d(sigma, truncated)), False)
         self.spatial_dims = spatial_dims
-        self.conv_n = [F.conv1d, F.conv2d, F.conv3d][spatial_dims - 1]
-        self.padding = same_padding(self.kernel.size()[0])
+        _sigma = ensure_tuple_rep(sigma, spatial_dims)
         self.device = device
-
-        self.kernel = self.kernel.to(self.device)
+        self.kernel = [
+            torch.nn.Parameter(torch.from_numpy(gaussian_1d(s, truncated)).to(self.device), False) for s in _sigma
```

Comment:
Maybe also consider to use `as_tensor(xxx, device=self.device)` here?

Reply:
ok, I changed, but what's the benefit of as_tensor?

Reply:
Thanks for your update.
There are 3 small benefits:
1. Using `torch.as_tensor(xxx, device)` is slightly simple.
2. `as_tensor()` can support not only numpy data, and python doesn't check data type, so always using `as_tensor()` can decrease potential data type related bugs.
3. `as_tensor()` has more sanity checks, like data contiguous, safer to use it.

Thanks.

Reply:
thanks @Nic-Ma, and you're right the nn.module interface makes it easier to use